shell/platform/android/platform_view_android_jni_impl.cc

```diff
@@ -508,6 +512,113 @@ static jboolean FlutterTextUtilsIsRegionalIndicator(JNIEnv* env,
                                                     jint codePoint) {
   return u_hasBinaryProperty(codePoint, UProperty::UCHAR_REGIONAL_INDICATOR);
 }
+
+static void LoadLoadingUnitFailure(intptr_t loading_unit_id,
+                                   std::string message,
+                                   bool transient) {
+  // TODO(garyq): Implement
+}
+
+static void DynamicFeatureInstallFailure(JNIEnv* env,
+                                         jobject obj,
+                                         jint jLoadingUnitId,
+                                         jstring moduleName,
+                                         jstring jError,
+                                         jboolean jTransient) {
+  LoadLoadingUnitFailure(static_cast<intptr_t>(jLoadingUnitId),
+                         fml::jni::JavaStringToString(env, jError),
+                         static_cast<bool>(jTransient));
+}
+
+static void LoadDartDeferredLibrary(JNIEnv* env,
+                                    jobject obj,
+                                    jlong shell_holder,
+                                    jint jLoadingUnitId,
+                                    jobjectArray jSearchPaths) {
+  // Convert java->c++
+  intptr_t loading_unit_id = static_cast<intptr_t>(jLoadingUnitId);
+  std::vector<std::string> search_paths =
+      fml::jni::StringArrayToVector(env, jSearchPaths);
+
+  // TODO: Switch to using the NativeLibrary class, eg:
+  //
+  //      fml::RefPtr<fml::NativeLibrary> native_lib =
+  //          fml::NativeLibrary::Create(lib_name.c_str());
+  //
+  // Find and open the shared library.
+  void* handle = nullptr;
+  while (handle == nullptr && !search_paths.empty()) {
+    std::string path = search_paths.back();
+    handle = ::dlopen(path.c_str(), RTLD_NOW);
+    search_paths.pop_back();
+    if (handle == nullptr) {
+      FML_LOG(ERROR) << "No dart shared library found at \"" << path
```

Comment:
are we expecting failures for the first entries of the searchpaths? i.e. is this spammy?

Reply:
We do expect some failures, I'll remove this log.